plugins/cairo-lang-macro-stable/src/ffi.rs

```diff
@@ -0,0 +1,44 @@
+/// This struct encapsulates a stable ABI representation of struct.
+///
+/// Please be aware that the memory management of values passing the FFI-barrier is tricky.
+#[repr(C)]
+#[derive(Debug)]
+#[doc(hidden)]
+pub struct StableSlice<T> {
+    ptr: *mut T,
+    len: usize,
+}
+
+impl<T> Copy for StableSlice<T> {}
+
+impl<T> Clone for StableSlice<T> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<T> StableSlice<T> {
+    /// Create a new `StableSlice` from a Vector.
+    /// Note that the vector will not be deallocated automatically.
+    /// Please make sure to use `into_owned` afterward, to free the memory.
+    pub fn new(mut x: Vec<T>) -> Self {
+        x.shrink_to_fit();
+        assert_eq!(x.len(), x.capacity());
+        let ptr = x.as_mut_ptr();
+        let len = x.len();
+        std::mem::forget(x);
+        Self { ptr, len }
+    }
+
+    /// Convert to owned vector.
+    pub fn into_owned(self) -> Vec<T> {
+        unsafe { Vec::from_raw_parts(self.ptr, self.len, self.len) }
+    }
+
+    /// Returns raw pointer and length.
+    /// Can be used to construct a slice.
+    /// No ownership is transferred.
+    pub fn into_raw_parts(self) -> (*mut T, usize) {
+        (self.ptr, self.len)
+    }
```

Comment:
I am not sure what is the purpose of this function. Wouldn't be possible to have `into_vec` instead? In this scenario seems cleaner to have function `new` and its inverse. Probably would require to ensure T can be converted from stable as well, but seems like desired property.

Reply:
I use it to construct a slice from raw pointer, it could return a slice already, but lifetimes management gets tricky then and I just kind of did this as a workaround 😅

Reply:
I wouldn't bother, this is not user facing API